src/Kestrel.Core/Internal/Http2/Http2Connection.cs

```diff
@@ -93,6 +93,8 @@ public Http2Connection(Http2ConnectionContext context)
             _serverSettings.HeaderTableSize = (uint)context.ServiceContext.ServerOptions.Limits.Http2.HeaderTableSize;
             _hpackDecoder = new HPackDecoder((int)_serverSettings.HeaderTableSize);
             _serverSettings.MaxHeaderListSize = (uint)context.ServiceContext.ServerOptions.Limits.MaxRequestHeadersTotalSize;
+            _serverSettings.InitialWindowSize = (uint)context.ServiceContext.ServerOptions.Limits.Http2.InitialWindowSize;
+            _inputFlowControl = new InputFlowControl(_serverSettings.InitialWindowSize, _serverSettings.InitialWindowSize / 2);
```

Comment:
nit: so far, it looks like `minWindowSizeIncrement` is always just `initialWindowSize/2`. Why don't we just do that calculation in the constructor instead of passing in two variables.

Reply:
We're pretty sure that will need to change once we start doing some testing. We'll need to play with it to find a reasonable balance.